File: src/MaksIT.WebUI/src/models/account/PostAccountRequest.ts
import z, { array, boolean, object, string, type ZodType } from 'zod'
import { ChallengeType } from '../../entities/ChallengeType'
import { RequestModelBase, RequestModelBaseSchema } from '@maks-it.com/webui-contracts'

export interface PostAccountRequest extends RequestModelBase {
  description: string
  contacts: string[]
  challengeType: string
  hostnames: string[]
  isStaging: boolean
  agreeToS: boolean
}

export const PostAccountRequestSchema: ZodType<PostAccountRequest> = RequestModelBaseSchema.and(
  object({
    description: string(),
    contacts: array(string()),
    hostnames: array(string()),
    challengeType: z.enum(ChallengeType),
    isStaging: boolean(),
    agreeToS: boolean()
  })
)

File: src/MaksIT.CertsUI.Client/Models/PostAccountRequest.cs
namespace MaksIT.CertsUI.Client.Models;

public class PostAccountRequest {
  public required string Description { get; set; }
  public required string[] Contacts { get; set; }
  public required string ChallengeType { get; set; }
  public required string[] Hostnames { get; set; }
  public required bool IsStaging { get; set; }
  public required bool AgreeToS { get; set; }
}
